### PR TITLE
Borg Hypospray Change

### PR DESCRIPTION
**Pull request by printer16**
This makes the borg hypospray take 3 seconds to inject someone.

Associated thread: https://forums.aurorastation.org/viewtopic.php?f=18&t=8197

### CODE REVIEW

**Comment by Lohikar** [comment]
Looks like you might need to merge dev into your branch.

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 1]
```diff
@@ -70,8 +70,10 @@
 			return
 
 	if (M.can_inject(user, 1))
-		user << "<span class='notice'>You inject [M] with the injector.</span>"
-		M << "<span class='notice'>You feel a tiny prick!</span>"
+		visible_message("<span class='notice'>[user] starts to inject [M] with their hypospray!</span>")
+		if(do_after(user, 3 SECONDS, act_mob = M))
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/borghydro.dm]: I wonder, check the regular syringes (human ones). See if they use `do_after` or `do_mob`. There's a nuanced difference and it should mimic this.

Comment (printer16) [inline code/modules/reagents/reagent_containers/borghydro.dm]: I thought it was act_mob. Whoops.

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 2]
```diff
@@ -70,8 +70,10 @@
 			return
 
 	if (M.can_inject(user, 1))
-		user << "<span class='notice'>You inject [M] with the injector.</span>"
-		M << "<span class='notice'>You feel a tiny prick!</span>"
+		visible_message("<span class='notice'>[user] starts to inject [M] with their hypospray!</span>")
+		if(do_mob(user, 3 SECONDS, act_target = M))
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/borghydro.dm]: Wrong param order:
`/proc/do_mob(mob/user , mob/target, delay = 30, numticks = 10, needhand = TRUE, display_progress = TRUE)`

Should be: `do_mob(user, M, 3 SECONDS)`

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 3]
```diff
@@ -70,8 +70,10 @@
 			return
 
 	if (M.can_inject(user, 1))
-		user << "<span class='notice'>You inject [M] with the injector.</span>"
-		M << "<span class='notice'>You feel a tiny prick!</span>"
+		visible_message("<span class='notice'>[user] starts to inject [M] with their hypospray!</span>")
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/borghydro.dm]: `mob/visible_message` works differently from other visible_message protos. It has a 2nd argument which is the same as going `user << ""`. So this should be changed to:
`visible_message("<span class='notice'>[user] starts to inject [M] with their hypospray!</span>", "<span class='notice'>You start to inject [M] with the injector.</span>")`

**Comment by LordFowl** [comment]
Isn't the point of a hypospray is that it is instant? What am I missing here?

**Comment by printer16** [comment]
Renamed to cyborg chemical injector.

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 4]
```diff
@@ -70,8 +70,10 @@
 			return
 
 	if (M.can_inject(user, 1))
-		user << "<span class='notice'>You inject [M] with the injector.</span>"
-		M << "<span class='notice'>You feel a tiny prick!</span>"
+		visible_message("<span class='notice'>[user] starts to inject [M] with their hypospray!</span>", "<span class='notice'>You start to inject [M] with your hypospray!</span>")
+		if(do_mob(user, M, 3 SECONDS))
```
Comment (SonicGotNuked) [inline code/modules/reagents/reagent_containers/borghydro.dm]: I think you should just keep the borg hypospray instant and instead remove soporific from the thing.

**Pull request merged**